pkg/logproto/compat.go

```diff
@@ -0,0 +1,267 @@
+package logproto
+
+import (
+	stdjson "encoding/json"
+	"fmt"
+	"math"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+	"unsafe"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/exemplar"
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/textparse"
+
+	"github.com/cortexproject/cortex/pkg/util"
+)
+
+// ToWriteRequest converts matched slices of Labels, Samples and Metadata into a WriteRequest proto.
+// It gets timeseries from the pool, so ReuseSlice() should be called when done.
+func ToWriteRequest(lbls []labels.Labels, samples []Sample, metadata []*MetricMetadata, source WriteRequest_SourceEnum) *WriteRequest {
+	req := &WriteRequest{
+		Timeseries: PreallocTimeseriesSliceFromPool(),
+		Metadata:   metadata,
+		Source:     source,
+	}
+
+	for i, s := range samples {
+		ts := TimeseriesFromPool()
+		ts.Labels = append(ts.Labels, FromLabelsToLabelAdapters(lbls[i])...)
+		ts.Samples = append(ts.Samples, s)
+		req.Timeseries = append(req.Timeseries, PreallocTimeseries{TimeSeries: ts})
+	}
+
+	return req
+}
```

Comment:
```suggestion

```

Reply:
Hmmm I think this is used indirectly:

- `ToWriteRequest` is invoked by `PusherAppender.Commit()`.
  -  `PusherAppender` is instantiated during `PusherAppendable.Appender()`
    - `PusherAppendable` is instantiated under `defaultTenantManagerFactory`
      - `defaultTenantManagerFactory` is invoked by `buildRuler()`
        - `buildRuler()` is used by many tests under `ruler/base`

Do you have a suggestion on how to get rid of it?